src/test/java/com/google/api/codegen/testdata/go_library.baseline

```diff
@@ -462,166 +462,153 @@ func (client *Client) ListStrings(ctx context.Context, req *google_example_libra
     var callOpts []gax.CallOption
     callOpts = append(callOpts, client.callOptions["ListStrings"]...)
     callOpts = append(callOpts, opts...)
-    atLastPage := false
-    return &StringIterator{
-        apiCall: func() (StringPage, error) {
-            if atLastPage {
-                return StringPage{}, io.EOF
-            }
-            var resp *google_example_library_v1.ListStringsResponse
-            err := gax.Invoke(ctx, func (c context.Context) error {
-                var err error
-                resp, err = client.client.ListStrings(c, req)
-                return err
-            }, callOpts...)
-            if err != nil {
-                return StringPage{}, err
-            }
-            if resp.NextPageToken == "" {
-                atLastPage = true
-            } else {
-                req.PageToken = resp.NextPageToken
-            }
-            return StringPage{
-                Items: resp.Strings,
-                NextPageToken: resp.NextPageToken,
-            }, nil
-        },
-    }, nil
+
+    iterator := &StringIterator{}
+    iterator.apiCall = func() error {
+        if iterator.atLastPage {
+            return Done
+        }
+        var resp *google_example_library_v1.ListStringsResponse
+        err := gax.Invoke(ctx, func (c context.Context) error {
+            var err error
+            req.PageToken = iterator.nextPageToken
```

Comment:
Would appreciate a sanity check on this `nextPageToken` logic. It didn't seem there was a pathway before for the user to actually provide a value for this - I included a way via the `SetPageToken` method.